save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityProjectWindow.kt

```diff
@@ -21,100 +22,203 @@ import react.useState
+                    setVulnerabilityProject(vulnerabilityProject.copy(`package` = newPackage))
+                    setConflictErrorMessage(null)
+                }
+            }
+
+            inputTextFormOptional {
+                form = InputTypes.PURL
+                textValue = vulnerabilityProject.`package`?.purl ?: ""
+                classes = "col-6 pl-2 pr-2 mt-3 text-left"
+                name = "Purl"
+                onChangeFun = { event ->
+                    val newPackage = vulnerabilityProject.`package`?.copy(purl = event.target.value)
+                    setVulnerabilityProject(vulnerabilityProject.copy(`package` = newPackage))
+                }
+            }
+
+            div {
+                className = ClassName("col-6 pl-2 pr-2 mt-3 text-left")
+                +"Language".t()
+
+                selectorBuilder(
+                    vulnerabilityProject.`package`?.language ?: "",
+                    VulnerabilityLanguage.values().map { it.toString() },
+                    classes = "input-group form-control custom-select",
+                ) { event ->
+                    val newPackage = vulnerabilityProject.`package`?.copy(language = event.target.value)
+                    setVulnerabilityProject(vulnerabilityProject.copy(`package` = newPackage))
+                }
+            }
+
+            div {
+                className = ClassName("col-6 pl-2 pr-2 mt-3 text-left")
+                +"Type version"
+
+                selectorBuilder(
+                    range.type.name,
+                    RangeType.values().map { it.toString() },
+                    classes = "input-group form-control custom-select",
+                ) { event ->
+                    val entries = event.target.value
+                    val typeSelect = requireNotNull(RangeType.valueOf(entries))
+                    val newRange = range.copy(type = typeSelect)
+                    setRange(newRange)
+                    setVulnerabilityProject(vulnerabilityProject.copy(ranges = listOf(newRange)))
+                }
+            }
+
+            inputTextFormOptional {
+                form = InputTypes.VERSION
+                textValue = range.events.first().introduced
+                classes = "col-6 pl-2 pr-2 mt-3 text-left"
+                name = "Introduces version"
+                onChangeFun = { event ->
+                    val newRange = range.copy(events = listOf(range.events.first().copy(introduced = event.target.value)))
+                    setRange(newRange)
+                    setVulnerabilityProject(vulnerabilityProject.copy(ranges = listOf(newRange)))
+                }
+            }
+
+            inputTextFormOptional {
+                form = InputTypes.GIT_URL
+                textValue = vulnerabilityProject.`package`?.introducedCommits?.first() ?: ""
+                classes = "col-6 pl-2 pr-2 mt-3 text-left"
+                name = "Introduces Url commits"
```

Comment:
Please be consistent about the case of abreviations.

> URL

Reply:
Fixed